### ISSUE TITLE
[Feature Request] [Terrain] Cliffs Terrain Please

### ISSUE THREAD

**Issue by knyghtmare** [wesnoth/wesnoth#6631]
### Describe the desired feature

Cliffs have apparently been a greatly requested terrain by all map-makers, UMC authors, and maybe some mainline scenarios. Yet, they haven't ever been considered for adding to the game's terrain selection. Would be nice to have.

[Reference link 1](https://forums.wesnoth.org/viewtopic.php?p=291302#p291302)

![image](https://user-images.githubusercontent.com/5283677/164132883-e7691070-d5e3-4b7f-b6d3-aba9480d68d5.png)

**Comment by mesilliac** [comment]
As the original author of that screenshot i'd be happy if someone could pick this up and make it work. It was a long time ago, but i did post the files i had working at the time in another post:
https://forums.wesnoth.org/viewtopic.php?p=291945#p291945

I'm not sure if much has changed since then, but i imagine it wouldn't need much tweaking. As far as i recall i never finished the junctions and wasn't happy with the cliff graphics themselves (in particular the bottoms of the cliffs, as it's just reusing the chasm graphics which fade to black), but looking at it now i think it's probably quite workable as it was.

**Comment by mesilliac** [comment]
I tried to get it to work but the terrain system has changed since then and idk how the new system works.

Actually the old system was basically black magic to me as well.

**Comment by cooljeanius** [comment]
Do the new raised/lowered terrains count for this?

**Comment by knyghtmare** [comment]
> Do the new raised/lowered terrains count for this?

No, not at all. I tried those out and they are unsatisfactory. This stays open until the UMC community is given their desired request, and that is, cliffs.

**Comment by cooljeanius** [comment]
Another old forum thread on this topic that I've come across recently: https://forums.wesnoth.org/viewtopic.php?t=10081

**Comment by stevecotton** [comment]
@knyghtmare it would be helpful to add an explanation of why they're unsatisfactory, or a description of what cliffs should do that those terrains don't.

**Comment by knyghtmare** [comment]
> it would be helpful to add an explanation of why they're unsatisfactory, or a description of what cliffs should do that those terrains don't.

Wanted something like this:
![Map-Screenshot_005](https://user-images.githubusercontent.com/5283677/217739266-c0db611b-ffd5-400c-bb59-7f9205f5d316.jpg)

Expected: 
1. Cliffs are unwalkable and centred on the hex
2. Cliffs can link with each other (shown in screenshot)
3. https://forums.wesnoth.org/viewtopic.php?p=291945#p291945 (we wanted this)

New Gulch/Bluff are okay but they are something new entirely and not cliffs.
There's no terrain cost difference when moving between layers what not.
Embellishments cannot be used on them.

**Comment by cooljeanius** [comment]
> Another old forum thread on this topic that I've come across recently: [forums.wesnoth.org/viewtopic.php?t=10081](https://forums.wesnoth.org/viewtopic.php?t=10081)

One more: https://forums.wesnoth.org/viewtopic.php?p=380432#p380432

**Comment by stevecotton** [comment]
> Cliffs are unwalkable and centred on the hex

Perhaps I'm misunderstanding the map, but it looks as if it's mostly unwalkable, and only playable with flying units. Is it meant to be a playable map, or just a graphics demo?

**Comment by knyghtmare** [comment]
> Is it meant to be a playable map, or just a graphics demo?

it's from a scenario of mine. Player has mostly land units and a flying type too.

**Comment by RKTnoob** [comment]
Yeah, the current gulches and bluffs are weak, knyghtmare is on point, I would like to see proper cliffs added jut like these. Additionally, if possible, make them able to stack vertically to show even larger height differences.

**Comment by CelticMinstrel** [comment]
I think the key thing about cliffs that bluff/gulch misses is that they're essentially a _wall_. It's like if you put a line of cave wall in, except that it's drawn like one side of the wall is higher than the other. (And if it's not too hard, it would be nice if it only blocks vision when you're on the lower side… though I'm not sure Wesnoth has any way to support that kind of thing.)

**Comment by Pentarctagon** [comment]
If all that's needed is for cliffs to be treated like they're impassable, that's probably really easy to do?

**Comment by CelticMinstrel** [comment]
There are other things you might want to see with cliffs that are more complicated, so I wouldn't say that the content of my previous post is the _only_ thing. For example, I think it would be expected that units on opposite sides of a cliff can attack each other as if they were adjacent. Maybe only with ranged weapons, though. And probably with a penalty for the person at the bottom and/or a bonus for the person at the top.

So yeah, there are a lot of things you'd want to do to make proper cliffs… but the general idea that they're a like a wall is, I think, the main thing that bluff/gulch does not provide.

**Comment by Pentarctagon** [comment]
I would suspect that making a realistic cliff implementation would quickly become overly complicated.

**Comment by Dalas121** [comment]
IMO the core feature being requested is an impassable cliff that looks pretty.  A wall.  I don't think much is needed beyond that.

Attacks don't need any special implementation, since cliffs take up a hex and prevent units from being adjacent, so they can never attack each-other in the first place.

**Comment by Pentarctagon** [comment]
To clarify from what was said on Discord, what's being requested here is to add the cliffs terrain from the UMC Terrain Pack to mainline.

**Comment by knyghtmare** [comment]
> To clarify from what was said on Discord, what's being requested here is to add the cliffs terrain from the UMC Terrain Pack to mainline.

Yes, that has been the entire objective of this feature request and this won't be closed until said objective has been met.
 
> IMO the core feature being requested is an impassable cliff that looks pretty. A wall. I don't think much is needed beyond that.

I mean...not impassable

> Expected:

> Cliffs are unwalkable and centred on the hex
> Cliffs can link with each other (shown in screenshot)
